realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java

```diff
@@ -247,6 +251,86 @@ public void createObjectFromJson_dateAsString() throws JSONException {
         assertEquals(new Date(1000), obj.getColumnDate());
     }
 
+    @Test
+    public void createObjectFromJson_decimal128() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("columnDecimal128", new Decimal128(BigDecimal.TEN));
```

Comment:
Well you could write your own parser which produces a `JSONObject/array` it could be useful to add directly the Bson types if you know that the underlying consumer system can consume (parse them)